code.js

```diff
@@ -1,3 +1,54 @@
-function augmentingPath(graph, start, end) {
-    return [];
+function augmentingPath(graph, start, end){
+	fullPath=[]
+	visitedNodes=[]
+	keyValues=Object.keys(graph);
+	source=keyValues[0]
+	for (var i=0; i<Object.keys(graph).length; i++){
+		visitedNodes[keyValues[i]]= Infinity
+	}
+	visitedNodes[source]=0
+	returnedValue=AP(graph, start, end);
+	if (returnedValue==0)
+		return fullPath
+	else if (returnedValue==1)
+		return []
+	else if (returnedValue==2)
+		return []
 }
+
+function AP(graph, currentNode, goalNode){
+	fullPath.push(currentNode)
+	if (goalNode==currentNode){
+		return 0
+	}
+	allVisited=true;
+	visitedKeys=Object.keys(visitedNodes)
```

Comment:
Looks like you're accessing a variable from the other function here? I would pass it as an argument then.